src/gsi/m_berror_stats_reg.f90

```diff
@@ -1132,11 +1167,141 @@ subroutine read_howv_stats(nlat,nlon,npar,arrout)
             read(unit=lun34 ,rec=j) (arrout(i,j,i_npar), i=1,nlat)
          enddo
          close(unit=lun34)
+         if (print_verbose .and. mype .eq. 0) then         
+           print*,trim(adjustl(myname)), trim(filename(i_npar)) // ' is used for background error of howv.'
+         end if
 
       else 
-         print*,myname, trim(filename(i_npar)) // ' does not exist'
+         if (print_verbose .and. mype .eq. 0) then         
+           print*,trim(adjustl(myname)), trim(filename(i_npar)) // ' does not exist for static BE of howv.'
+           print*,trim(adjustl(myname)), '::using pre-defined corp_howv=',corp_howv, ' hwllp_howv=',hwllp_howv, ' for howv.'
+         end if
       end if
    end do
 end subroutine read_howv_stats
+!++++
+subroutine get_factor_tuneBE_howv(mype, b_factor)
+!$$$  subprogram documentation block
```

Comment:
If line 279-line 283 can move before line 243 (i.e, initialize the ensemble part before the static  part) in glbsoi.f90, then we don't need to read hyb_info file one more time here. I think the above two parts are independent, we can do regression tests to verify this.

Reply:
I did regressional tests for [this change](https://github.com/guoqing-noaa/GSI/commit/c9e30d41ebd16f0a42f0b091142d746ef43ba814) and all tests passed.

Reply:
@guoqing-noaa 
I agree with you. This is a good idea. And thank you very much for doing the ctest for this idea.

However, my concern is that, glbsoi.f90 is a very important subroutine in GSI. I am reluctant to modify the glbsoi.f90, such a main driver of GSI, just for howv, which is not a vital product of GSI. 

Thanks,

Reply:
@GangZhao-NOAA It should be fine to modify glbsoi.f90 if necessary and no harm. I suggest you go ahead to make the needed changes. If the changes are not allowed, the reviewers and the PR handling team will provide feedback. We should avoid unnecessary rereading of the hyb_info file.

Reply:
@guoqing-noaa 
I agree with you that the "unnecessary rereading of the hyb_info file" should be avoided. Based on the previous discussions with Manuel, he does not like the idea to tune/adjust the static BE of howv in GSI. He would like to simply set the tuned BE of howv in the namelist (using corp_hwllp), or directly set the tuned values in the binary file of static BE of howv.

So I am going to remove "subroutine get_factor_tuneBE_howv(mype, b_factor)" and the related code. This would simplify the code (no more print-out of corp of howv after the tuning), remove a namelist option l_tuneBE_howv, and then no need to modify glbsoi.f90.

Thanks,